### PR TITLE
Switch to  GH Actions

### PR DESCRIPTION
**Pull request by scytacki**
This uses GH Actions to build and deploy the code.
It saves code coverage stats to codecov.org
It saves cypress results to dashboard.cypress.io

The travis file that it replaces used to save code coverage to codeclimate.

Most of the changes here have been added as PRs to starter-projects as well:
https://github.com/concord-consortium/starter-projects/pull/27 - jest coverage in codecov.io
https://github.com/concord-consortium/starter-projects/pull/28 - cypress dashboard
https://github.com/concord-consortium/starter-projects/pull/29 - cypress coverage in codecov.io
https://github.com/concord-consortium/starter-projects/pull/30 - run cypress tests in parallel

This repository was already configured to record cypress coverage information, so the changes here are less than what was needed to add this cypress code coverage to starter-projects.

### CODE REVIEW

**Comment by scytacki** [comment]
Successful build using this config here:
https://github.com/concord-consortium/portal-report/runs/1433380978?check_suite_focus=true#step:7:24

File: s3_deploy.sh [hunk 1]
```diff
@@ -1,39 +1,99 @@
...
+# Typically this is the Project name
+# The portal report has its own bucket and deploys to the root of it
+S3_BUCKET_PREFIX=''
+# AWS CloudFront distribution ID
+DISTRIBUTION_ID='E19KBL7INSABNU'
+# AWS CloudFront distribution domain
+DISTRIBUTION_DOMAIN='portal-report.concord.org'
+# name of branch to deploy to root of site
+ROOT_BRANCH='production'
+# Bucket to deploy to, typically this is 'model-resourcs', but some projects
```
Comment (kswenson) [inline s3_deploy.sh:12]:
```suggestion
# Bucket to deploy to, typically this is 'model-resources', but some projects
```
This typo presumably exists in other instances of this file as well. How did I/we miss it before? 🤷

**Review by kswenson — APPROVED** [review]
👍 LGTM

File: s3_deploy.sh (hunk 1 above)
Comment (scytacki) [reply s3_deploy.sh:12]: Thanks. I just updated it in starter-projects too.

**Pull request merged**